basicsettings.ml

```diff
@@ -24,14 +24,35 @@ let concurrent_server = Settings.add_bool ("concurrent_server", true, `System)
 let printing_types = Settings.add_bool ("printing_types", true, `User)
 
 (** Name of the file containing the prelude code. *)
-let prelude_file = 
+let prelude_file =
   let prelude_dir = match Utility.getenv "LINKS_LIB" with
-      None -> Filename.dirname Sys.executable_name
+    (* If user defined LINKS_LIB then it takes the highest priority *)
     | Some path -> path
+    | None ->
+        (* If LINKS_LIB is not defined then we search in current directory *)
+        let current_dir = Filename.dirname Sys.executable_name in
+        if Sys.file_exists (Filename.concat current_dir "prelude.links") then
+          current_dir
+        else
+          try
+            (* If all else failed we search for OPAM installation of Links and
+               use a prelude that it provides *)
+            let opam_lib_dir =
+              input_line (Unix.open_process_in "opam config var lib 2>/dev/null") in
```

Comment:
We might want to use `opam config var links:lib`. It prints `"#undefined"` for me (I don't have links installed through opam). `opam config var ocamlbuild:lib` prints `"/home/stefan/.opam/system/lib/ocamlbuild"` for me. There's probably nobody who has different lib dirs from the defaults, but at least we save ourselves the concat .. "links" line.

(Not sure whether we need to do something in the opam file to populate the var links:lib thing.)